### ISSUE TITLE
assertion failed: piece < pieceCount()

### ISSUE THREAD

**Issue by beren12** [transmission/transmission#4352]
### What is the issue?

Crashing after a few hours with 1500 torrents

```2022-12-11 21:22:49.717 Transmission[21092:1310842] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'assertion failed: piece < pieceCount() (/Users/mac-5xbhtnjszu/workspace/trunk-mac/libtransmission/block-info.cc:71)'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff38db1a17 __exceptionPreprocess + 250
	1   libobjc.A.dylib                     0x00007fff71736a9e objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff38db1875 +[NSException raise:format:] + 181
	3   Transmission                        0x000000010019fc1c _Z16tr_assert_reportNSt3__117basic_string_viewIcNS_11char_traitsIcEEEEiS3_ + 294
	4   Transmission                        0x00000001001dc013 _ZNK13tr_block_info8pieceLocEjjj + 87
	5   Transmission                        0x000000010017ac2b _ZL16peerCallbackFuncP7tr_peerPK13tr_peer_eventPv + 536
	6   Transmission                        0x000000010018b9fd _ZL13readBtMessageP15tr_peerMsgsImplP8evbufferm + 10368
	7   Transmission                        0x0000000100183472 _ZL7canReadP9tr_peerIoPvPm + 2046
	8   Transmission                        0x00000001001709c7 _ZL14canReadWrapperP9tr_peerIo + 252
	9   Transmission                        0x000000010016be0d _ZL12utp_callbackP22utp_callback_arguments + 2723
	10  Transmission                        0x0000000100212625 _Z16utp_call_on_readP18struct_utp_contextP9UTPSocketPKhm + 49
	11  Transmission                        0x00000001002152cb _Z20utp_process_incomingP9UTPSocketPKhmb + 2840
	12  Transmission                        0x0000000100215e78 utp_process_udp + 485
	13  Transmission                        0x00000001001bf10a _Z12tr_utpPacketPKhmPK8sockaddrjP10tr_session + 126
	14  Transmission                        0x00000001001bd5ed _ZL14event_callbackisPv + 353
	15  Transmission                        0x00000001001fab4d event_process_active_single_queue + 1133
	16  Transmission                        0x00000001001f79c5 event_base_loop + 1130
	17  Transmission                        0x0000000100164999 _ZL18libeventThreadFuncP15tr_event_handle + 253
	18  Transmission                        0x00000001001655c5 _ZNSt3__1L14__thread_proxyINS_5tupleIJNS_10unique_ptrINS_15__thread_structENS_14default_deleteIS3_EEEEPFvP15tr_event_handleES8_EEEEEPvSC_ + 40
	19  libsystem_pthread.dylib             0x00007fff72adf109 _pthread_start + 148
	20  libsystem_pthread.dylib             0x00007fff72adab8b thread_start + 15
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

### Which application of Transmission?

macOS app

### Which version of Transmission?

5fc9f22fab

**Comment by ckerr** [comment]
@beren12 I haven't been able to reproduce this issue. Is it still happening for you?

**Comment by ckerr** [comment]
Fixed at last by #5097
